types/src/jwks/mod.rs

```diff
@@ -55,3 +77,35 @@ pub struct ObservedJWKs {
 pub struct PatchedJWKs {
     pub jwks: AllProvidersJWKs,
 }
+
+impl PatchedJWKs {
+    pub fn latest_version(&self, iss: &str) -> Option<&ProviderJWKs> {
+        let mut latest: Option<&ProviderJWKs> = None;
+        for provider_jwk_set in &self.jwks.entries {
+            if provider_jwk_set.issuer.eq(&issuer_from_str(iss)) {
```

Comment:
Is there a reason to keep the older version entries on-chain?

Reply:
cc @zjma

Reply:
the version is not supposed to be checked here.
An `AllProvidersJWKs` object (e.g, here self.jwks) is effectively a map and there will be no duplicated issuers.

Reply:
The version is there only for consensus purpose. Authenticator stuff doesn't need it.

Reply:
what's in `ObservedJWKs`/`PatchedJWKs` is always the latest version.

Reply:
Also I realized here we are not yet utilizing the Move helper function `get_patched_jwk()` we designed?

Reply:
d'oh, I looked for it in the move file and didn't find it.  ya I'd like to use the move helper

Reply:
resolved offline - will use the current resolver to bulk fetch all jwks and avoid fetching per authenticator